lib/spack/spack/fetch_strategy.py

```diff
@@ -980,8 +980,18 @@ def fetch(self):
 
         # Init submodules if the user asked for them.
         if self.submodules:
+            args = ['submodule', 'update', '--init']
+            if isinstance(self.submodules, (list, tuple)):
+                args.extend(self.submodules)
+            elif isinstance(self.submodules, six.string_types):
+                args.extend([_.strip() for _ in self.submodules.split(",") if _.split()])
+            elif callable(self.submodules):
+                # User defined function, allows specifying submodules on a per-variant
+                # basis
+                args.extend(self.submodules(self.package.spec))
+            else:
+                args.append("--recursive")
```

Comment:
Don't you always want this, as particularly chosen submodules might have submodules of their own? This change would make `submodules='external/foo'` different from `submodules=True`.